com.ca.lsp.cobol/lsp-core-cobol-parser/src/main/java/com/ca/lsp/core/cobol/preprocessor/sub/document/impl/CobolSemanticParserListenerImpl.java

```diff
@@ -134,9 +136,21 @@ public void exitCopyStatement(@Nonnull CopyStatementContext ctx) {
     String copybookName = retrieveCopybookName(copySource);
     Position position = retrievePosition(copySource);
     defineCopybook(copybookName, position);
+    checkCopybookNameLength(copybookName, position);
     this.preprocessorCleanerService.excludeStatementFromText(ctx, COMMENT_TAG, tokens);
   }
 
+  private void checkCopybookNameLength(String copybookName, Position position) {
+    if (copybookName != null && copybookName.length() > 8) {
+      errors.add(
+          SyntaxError.syntaxError()
+              .severity(3)
+              .suggestion(String.format(COPYBOOK_OVER_8_CHARACTERS, copybookName))
+              .position(position)
+              .build());
+    }
```

Comment:
I saw that this routine is very similar as the method ```toSyntaxError```
I think is a good idea try to define a more general function and call it two times or even better delegate this responsibility to an appropriate class and made it more general and available for all the core and service module

Reply:
This class is going to be deleted and its logic will be moved to `CobolSemanticParser`, so at least for me it is better to keep it like this and refactor later.

Reply:
ok